libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt

```diff
@@ -243,7 +243,9 @@ class PredictiveCacheController constructor(
      */
     fun removeMapControllers(map: MapboxMap) {
         mapListeners[map]?.let {
-            map.removeOnStyleLoadedListener(it)
+            if (map.isValid()) {
+                map.removeOnStyleLoadedListener(it)
```

Comment:
What if the map was destroyed, but the listener hadn't been removed? Would there be a leak in this case?

Reply:
Guys from maps team confirmed that listeners are removed automatically on destroy

Reply:
What about onDestroy? We also reference a map there to remove listeners.

Reply:
Added there as well, thank you